test/dict.jl

```diff
@@ -662,3 +662,11 @@ end
     @test f.x[2].first == (-, 6, 5, 3, 5, 8)
     @test f.x[2].second == 1
 end
+
+@testset "issue #18708 error type for dict constructor" begin
+    @test_throws UndefVarError Dict(x => y for x in 1:10)
+end
+
+@testset "issue #19179 throwing error in dict constructor" begin
+    @test_throws ErrorException Dict(i => error("foo") for i in 1:10)
```

Comment:
if the error that the issue was complaining about were an ErrorException (or changed to one) then this test wouldn't identify the problem